kpi/filters.py

```diff
@@ -175,6 +186,47 @@ def _get_queryset_for_collection_statuses(self, request, queryset):
 
         return queryset
 
+    def _get_queryset_for_discoverable_child_assets(
+        self, request: Request, queryset: QuerySet
+    ) -> QuerySet:
+        """
+        Returns a queryset containing the children of publically discoverable
+        assets based on the discoverability of the child's parent. The parent
+        uid is passed in the request query string.
+
+        args:
+            request (Request)
+            queryset (QuerySet)
+
+        returns:
+            QuerySet
+        """
+
+        self._return_queryset = False
+        PARENT_UID_PARAMETER = 'parent__uid'
+
+        if 'q' not in request.query_params:
+            return queryset
+
+        request_query = request.query_params['q']
+        parent_uid = re.search(
+            f'{PARENT_UID_PARAMETER}:([a-zA-Z0-9]*)', request_query
```

Comment:
Let's not use this as a long-term solution, but for now, I'm going to merge it into `public-collections-tweaks` so that we can do more front-end testing there

Reply:
It did feel very hacky to do it that way... definitely all for a better solution.